src/librustdoc/doctest.rs

```diff
@@ -382,9 +394,16 @@ pub(crate) fn run_tests(
     // `running 0 tests...`.
     if ran_edition_tests == 0 || !standalone_tests.is_empty() {
         standalone_tests.sort_by(|a, b| a.desc.name.as_slice().cmp(b.desc.name.as_slice()));
-        test::test_main(&test_args, standalone_tests, None);
+        let temp_dir_path = temp_dir.as_ref().map(|t| t.path());
+        test::test_main_with_exit_callback(&test_args, standalone_tests, None, || {
+            // We cannot move `temp_dir` in this closure so instead we do its job manually.
+            if let Some(temp_dir_path) = temp_dir_path {
+                let _ = std::fs::remove_dir_all(temp_dir_path);
+            }
+        });
```

Comment:
Can't you replace it with `None?

```suggestion
        test::test_main_with_exit_callback(&test_args, standalone_tests, None, || {
            // We need to drop 
            std::mem::drop(temp_dir.take());
        });
```